tools/tplg_parser/tplg_parser.c

```diff
@@ -1193,51 +1201,62 @@ int load_widget(void *dev, int dev_type, struct comp_info *temp_comp_list,
 		if (load_pipeline(dev, comp_id, pipeline_id, widget,
 				  *sched_id) < 0) {
 			fprintf(stderr, "error: load pipeline\n");
-			return -EINVAL;
+			ret = -EINVAL;
+			goto err;
 		}
 		break;
 	case(SND_SOC_TPLG_DAPM_SRC):
 		if (load_src(dev, comp_id, pipeline_id, widget, tp) < 0) {
 			fprintf(stderr, "error: load src\n");
-			return -EINVAL;
+			ret = -EINVAL;
+			goto err;
 		}
 		break;
 	case(SND_SOC_TPLG_DAPM_ASRC):
 		if (load_asrc(dev, comp_id, pipeline_id, widget, tp) < 0) {
 			fprintf(stderr, "error: load src\n");
-			return -EINVAL;
+			ret = -EINVAL;
+			goto err;
 		}
 		break;
 	case(SND_SOC_TPLG_DAPM_MIXER):
 		if (load_mixer(dev, comp_id, pipeline_id, widget) < 0) {
 			fprintf(stderr, "error: load mixer\n");
-			return -EINVAL;
+			ret = -EINVAL;
+			goto err;
 		}
 		break;
 	case(SND_SOC_TPLG_DAPM_EFFECT):
 		if (load_process(dev, comp_id, pipeline_id, widget) < 0) {
 			fprintf(stderr, "error: load effect\n");
-			return -EINVAL;
+			ret = -EINVAL;
+			goto err;
 		}
 		break;
 	/* unsupported widgets */
 	default:
 		if (fseek(file, widget->priv.size, SEEK_CUR)) {
 			fprintf(stderr, "error: fseek unsupported widget\n");
-			return -errno;
+			ret = -errno;
+			goto err;
 		}
 
 		printf("info: Widget type not supported %d\n", widget->id);
 		ret = tplg_load_controls(widget->num_kcontrols, file);
 		if (ret < 0) {
 			fprintf(stderr, "error: loading controls\n");
-			return ret;
+			goto err;
 		}
 		break;
 	}
 
 	free(widget);
 	return 0;
+
+err:
```

Comment:
minor nit, you could set to ret = 0 before the label so you don't duplicate the exit path

Reply:
Agree. But i changed the label from 'err' to 'exit' to not cause any confusion. (As in case of no error too we would be entering the label)